js/navbar/tabColor.js

```diff
@@ -207,33 +181,16 @@ var runNetwork = function anonymous (input) {
   return output
 }
 
-window.updateColorPalette = function () {
-  var tab = tabs.get(tabs.getSelected())
-
-  if (tab.private) {
-    // private tabs have their own color scheme
-    return setColor(defaultColors.private[0], defaultColors.private[1])
-  // use the colors extracted from the page icon
-  } else if (tab.backgroundColor || tab.foregroundColor) {
-    return setColor(tab.backgroundColor, tab.foregroundColor)
-  // otherwise use the default colors
-  } else if (window.isDarkMode) {
-    return setColor(defaultColors.darkMode[0], defaultColors.darkMode[1])
-  } else {
-    return setColor(defaultColors.lightMode[0], defaultColors.lightMode[1])
-  }
-}
-
 function setColor (bg, fg) {
-  var background = document.getElementsByClassName('theme-background-color')
-  var textcolor = document.getElementsByClassName('theme-text-color')
+  var backgroundElements = document.getElementsByClassName('theme-background-color')
+  var textElements = document.getElementsByClassName('theme-text-color')
 
-  for (var i = 0; i < background.length; i++) {
-    background[i].style.backgroundColor = bg
+  for (var i = 0; i < backgroundElements.length; i++) {
+    backgroundElements[i].style.backgroundColor = bg
```

Comment:
An alternative style:

    backgroundElements.forEach(el => { el.style.backgroundColor = bg })
    textElements.forEach(el => { el.style.color = fg })

Just sayin'

Reply:
Or with the traditional way, at least don't use `var i =…` for iteration, because it makes a function-scoped variable `i`, which is [might be] also used in any further loops. Use `let` instead.

Reply:
getElementsByClassName returns an `HTMLCollection` that doesn't have a `forEach` method, so it would be something like `Array.prototype.forEach.call(backgroundElements, el => ...`. Maybe it's still worth it?

(I guess we could use querySelectorAll instead, but that seems kind of pointless.)